File: src/main/java/io/github/kingg22/api/vacunas/panama/persistence/repository/DireccionRepository.java
package io.github.kingg22.api.vacunas.panama.persistence.repository;

import io.github.kingg22.api.vacunas.panama.persistence.entity.Direccion;
import java.util.List;
import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;

public interface DireccionRepository extends JpaRepository<Direccion, UUID> {

    Optional<List<Direccion>> findDireccionByDireccionAndDistrito_Id(String direccion, int idDistrito);

    Optional<List<Direccion>> findDireccionByDireccionStartingWith(String direccion);

    Optional<List<Direccion>> findDireccionByDireccionAndDistrito_Nombre(String direccion, String distrito);
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/persistence/repository/DireccionRepository.kt
@file:Suppress("ktlint:standard:function-naming", "kotlin:S100")

package io.github.kingg22.api.vacunas.panama.persistence.repository

import io.github.kingg22.api.vacunas.panama.persistence.entity.Direccion
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional
import java.util.UUID

interface DireccionRepository : JpaRepository<Direccion, UUID> {
    fun findDireccionByDireccionAndDistrito_Id(direccion: String?, idDistrito: Int): Optional<List<Direccion>>

    fun findDireccionByDireccionStartingWith(direccion: String?): Optional<List<Direccion>>

    fun findDireccionByDireccionAndDistrito_Nombre(direccion: String, distrito: String?): Optional<List<Direccion>>
}
